sql/hive/src/test/scala/org/apache/spark/sql/hive/StatisticsSuite.scala

```diff
@@ -1416,4 +1416,21 @@ class StatisticsSuite extends StatisticsCollectionTestBase with TestHiveSingleto
       assert(catalogStats.rowCount.isEmpty)
     }
   }
+
+  test(s"CTAS should update statistics if ${SQLConf.AUTO_SIZE_UPDATE_ENABLED.key} is enabled") {
```

Comment:
In addition, please add the test coverage at `sql/core/src/test/scala/org/apache/spark/sql/StatisticsCollectionSuite.scala`, too.

Reply:
Please review this PR first: https://github.com/apache/spark/pull/24596